src/plugins/clips-executive/clips/goals/run-parallel.clp

```diff
@@ -0,0 +1,153 @@
+;---------------------------------------------------------------------------
+;  run-parallel.clp - CLIPS executive - goal to run all sub-goals in parallel
+;
+;  Created: Wed Dec 16 2020
+;  Copyright  2020  Tarik Viehmann
+;  Licensed under GPLv2+ license, cf. LICENSE file
+;---------------------------------------------------------------------------
+
+; Sub-type: RUN-SUBGOALS-IN-PARALLEL
+; Perform: all sub-goals in parallel
+; Params: (params continue-on FAILED|REJECTED) to not stop SELECTION of
+;         other goals when a sub-goal FAILED/is REJECTED
+;         other values will be ignored
+; Succeed: if all sub-goal succeeds
+; Fail:    if at least one sub-goal fails
+; Reject:  if any sub-goal is rejected but no sub-goal failed
+;
+; A RUN-PARALLEL parent goal will run all sub-goals in parallel, starting
+; then ordered by their priorities.
+; If any goal fails, the parent fails. If any sub-goal is rejected,
+; the parent is rejected. If all goals have been completed successfully,
+; the parent goal succeeds.
+; The RUN-PARALLEL goal can be configured to continue execution of other goals
+; upon encountering a FAILED or REJECTED sub-goal by including
+; 'continue-on FAILED' or 'continue-on REJECTED' in the goal params
+; (continue-on FAILED implies continue-on REJECTED).
+
+;
+; Interactions:
+; - User FORMULATES goal
+; - User SELECTS goal
+; - User EXPANDS goal, consisting of:
+;   * create goals with parent ID equal the RUN-PARALLEL goal ID
+;   * set RUN-PARALLEL goal mode to EXPANDED
+; - Automatic: if no sub-goal formulated -> FAIL
+; - Automatic: if all sub-goals rejected -> REJECT
+; - Automatic: DISPATCH all formulated sub-goals with highest priority by
+;              SELECTING them- SELECT next batch of sub-goals once all
+;              previously SELECTED subgoals are past COMMITTMENT stage.
```

Comment:
By "past COMMITTMENT" you mean not COMMITTED anymore, right?
```suggestion
;              previously SELECTED subgoals are DISPATCHED or FINISHED
```

Reply:
Yea, improved the wording now.